lib/deploy/functions/deploy.js

```diff
@@ -1,19 +1,23 @@
 'use strict';
 
+var _ = require('lodash');
 var chalk = require('chalk');
 var RSVP = require('rsvp');
 var tmp = require('tmp');
 var utils = require('../../utils');
 var gcp = require('../../gcp');
 var prepareFunctionsUpload = require('../../prepareFunctionsUpload');
 
+var GCP_REGION = 'us-central1';
+
 tmp.setGracefulCleanup();
 
 module.exports = function(context, options, payload) {
   var _uploadSource = function(source) {
-    return gcp.storage.buckets.acquire(context.projectId, context.functionsBucket)
-    .then(function(bucketName) {
-      return gcp.storage.upload(source.stream, bucketName);
+    return gcp.cloudfunctions.generateUploadUrl(context.projectId, GCP_REGION).then(function(uploadUrl) {
+      _.set(context, 'uploadUrl', uploadUrl);
+      uploadUrl = _.replace(uploadUrl, 'https://storage.googleapis.com', '');
```

Comment:
`_.trimPrefix(uploadUrl, 'https://storage.googleapis.com)`

Reply:
I think that's a golang method?

Reply:
Sorry [`_.trimStart`](https://lodash.com/docs/4.17.5#trimStart)